app/stores/toplevel/TransactionsStore.js

```diff
@@ -467,22 +469,69 @@ export default class TransactionsStore extends Store {
   }
 
   exportTransactionsToFile: {|
-    publicDeriver: IPublicDeriver<ConceptualWallet & IHasLevels>,
+    publicDeriver: PublicDeriver<ConceptualWallet>,
     exportRequest: TransactionRowsToExportRequest,
   |} => Promise<void => Promise<void>> = async (request) => {
     const txStore = this.stores.transactions;
-    const respTxRows = [];
+    let respTxRows = [];
 
     const apiType = getApiForNetwork(request.publicDeriver.getParent().getNetworkInfo());
+    const delegationStore = this.stores.delegation;
+    const delegationRequests = delegationStore.getDelegationRequests(request.publicDeriver);
 
     await txStore.getTransactionRowsToExportRequest.execute(async () => {
+      const selectedNetwork = request.publicDeriver.getParent().getNetworkInfo();
+      const withLevels = asHasLevels<ConceptualWallet>(request.publicDeriver);
+      if (!withLevels) return;
       const rows = await this.api[apiType].getTransactionRowsToExport({
-        publicDeriver: request.publicDeriver,
+        publicDeriver: withLevels,
         ...request.exportRequest,
       });
+
+      /**
+       * NOTE: The rewards export currently supports only Haskell Shelley
+       */
+      if (isCardanoHaskell(selectedNetwork) && delegationRequests) {
+        const rewards = await delegationRequests.rewardHistory.promise;
+        if (rewards != null) {
+          const apiMeta = getApiMeta(
+            getApiForNetwork(selectedNetwork)
+          )?.meta;
+          if (apiMeta == null) throw new Error(`${nameof(TransactionsStore)} no API selected`);
+          const amountPerUnit = new BigNumber(10).pow(apiMeta.decimalPlaces);
+          const fullConfig = getCardanoHaskellBaseConfig(selectedNetwork);
+
+          const absSlotFunc = await timeUtils.genToAbsoluteSlotNumber(fullConfig);
+          const timeSinceGenFunc = await timeUtils.genTimeSinceGenesis(fullConfig);
+          const realTimeFunc = await timeUtils.genToRealTime(fullConfig);
+          const rewardRows = rewards.map(item => {
+            const absSlot = absSlotFunc({
+              epoch: item[0],
+              slot: 0
+            });
+            const epochStartDate = realTimeFunc({
+              absoluteSlotNum: absSlot,
+              timeSinceGenesisFunc: timeSinceGenFunc
+            });
+            return {
+              type: 'in',
+              amount: item[1].div(amountPerUnit).toString(),
+              fee: '0',
+              date: epochStartDate,
+              comment: 'Staking Reward'
```

Comment:
I think it could be helpful if we add the Epoch number as well

Reply:
done